### PR TITLE
Add Journal Notes

### PR DESCRIPTION
**Pull request by alx-ef**

### CODE REVIEW

File: internal/fs/fs.go [hunk 1]
```diff
@@ -127,6 +128,10 @@ func (fs FS) Put(dir, filename, content string) error {
 		return fmt.Errorf("put: unsafe path '%s': %w", path, errUnsafePath)
 	}
 
+	dirs := strings.Split(path, string(os.PathSeparator))
+	dirs = dirs[:len(dirs)-1]
+	os.MkdirAll(strings.Join(dirs, string(os.PathSeparator)), 0755)
```
Comment (zakirullin) [inline internal/fs/fs.go]: fs.backend.MkdirAll
+ error handling

File: internal/journal/journal.go [hunk 2]
```diff
@@ -1,24 +1,57 @@
...
 const (
-	dateFormat  = "02, Monday"
-	headerLevel = 4
+	headerLevel        = 4
+	intraNoteSeparator = "; "
 )
 
-var now = time.Now // to be replaced in tests
+func AddDailyNote(dir, noteFilename string, botFs *fs.FS, journalFilenameFormat, journalHeaderFormat string) error {
+	// TODO: somehow lock the file
```
Comment (zakirullin) [inline internal/journal/journal.go]: I thought about this too. 

I believe that for the sake of simplicity we would introduce global lock by user id

So requests from one user are gonna be processed sequentially. 

Actually that's the way it works in the current bot, and we had no problems whatsoever. 

I.e. we can write code thinking that there are no parallel routines for the same user.

File: internal/journal/journal.go [hunk 3]
```diff
@@ -1,24 +1,57 @@
...
+	content, err := botFs.Content(dir, noteFilename)
+	if err != nil {
+		return fmt.Errorf("failed to move to journal: can't get note content: %w", err)
+	}
+	note := fs.Title(noteFilename)
+	if strings.TrimSpace(content) != "" {
+		for _, line := range strings.Split(content, "\n") {
```
Comment (zakirullin) [inline internal/journal/journal.go:30]: We can use `text.NormNewLines()` to handle all kind of different platform-dependent newlines

Comment (alx-ef) [reply internal/journal/journal.go:30]: there is on such a method. Also I checked the go's std lib, they just use \n https://go.dev/src/fmt/print.go#:~:text=p.buf.writeByte(%27%5Cn%27)
It's pretty standard these days, I use \n even when working with Windows

File: internal/fs/fs.go (hunk 1 above)
Comment (alx-ef) [reply internal/fs/fs.go]: done

File: internal/journal/journal.go (hunk 2 above)
Comment (alx-ef) [reply internal/journal/journal.go]: Anyway, it's out of the scope of this PR. I removed the comment.
I'll prepare a new PR, with a channel-based solution for this issue once we merge this one. I believe it's more idiomatic for Go than using locks)

File: internal/journal/journal.go (hunk 3 above)
Comment (zakirullin) [reply internal/journal/journal.go:30]: Yes, it is used for writing, but when we are reading - a md file might contain `\r\n`, for example.

I myself had a few MD files that were created in some 3rd party app, and the bot didn't handle them correctly when I split by "\n".

NormLineEndings function is here:
`pkg/text/str.go:49`

You can search through golang's stdlib source code, and `\r\n` is still being taken care of upon reading:
`// Carriage-return followed by newline. (We treat \r\n as \n always.`

```
	// line comment (does not include newline)
	// (on Windows, the line comment may end in \r\n)
	i := len(s)
	if s[i-1] == '\r' {
		i--
```

`// Trim is line without \n or \r\n`

Comment (zakirullin) [reply internal/journal/journal.go:30]: It's not a blocking issue, I'll merge to avoid stale branches

**Pull request merged**